src/compiler/checker.ts

```diff
@@ -1477,6 +1477,15 @@ namespace ts {
                         if (location.parent && isClassElement(location.parent)) {
                             location = location.parent;
                         }
+                        //
+                        //  type T = number;
+                        //  @decorator((arg: T) => {}) // <-- type T should be resolved in the container, not the class
+                        //  class C<T> {
+                        //  }
+                        //
+                        if (location.parent && isClassLike(location.parent)) {

```

Comment:
My fix in #28609 works a bit different for decorators on class members, see the added test.
Could you please verify which is the desired behavior.

Reply:
Ah, my apologies. I did not notice there was an existing PR for this. Your logic in #28609 looks solid, considering that we do not currently support decorators on class expressions. I will close this PR in favor of yours.